crates/database/src/jit_orders.rs

```diff
@@ -0,0 +1,256 @@
+            partially_fillable = EXCLUDED.partially_fillable,
+            signature = EXCLUDED.signature,
+            receiver = EXCLUDED.receiver,
+            signing_scheme = EXCLUDED.signing_scheme,
+            sell_token_balance = EXCLUDED.sell_token_balance,
+            buy_token_balance = EXCLUDED.buy_token_balance;
+        "#,
+    );
+
+    let query = query_builder.build();
+    query.execute(ex).await?;
+
+    Ok(())
+}
+
+impl From<FullJitOrder> for orders::FullOrder {
+    fn from(jit_order: FullJitOrder) -> Self {
+        orders::FullOrder {
+            uid: jit_order.uid,
+            owner: jit_order.owner,
+            creation_timestamp: jit_order.creation_timestamp,
+            sell_token: jit_order.sell_token,
+            buy_token: jit_order.buy_token,
+            sell_amount: jit_order.sell_amount,
+            buy_amount: jit_order.buy_amount,
+            valid_to: jit_order.valid_to,
+            app_data: jit_order.app_data,
+            fee_amount: jit_order.fee_amount.clone(),
+            full_fee_amount: jit_order.fee_amount,
+            kind: jit_order.kind,
+            class: orders::OrderClass::Limit, // irrelevant
+            partially_fillable: jit_order.partially_fillable,
+            signature: jit_order.signature,
+            sum_sell: jit_order.sum_sell,
+            sum_buy: jit_order.sum_buy,
+            sum_fee: jit_order.sum_fee,
+            invalidated: false,
+            receiver: Some(jit_order.receiver),
+            signing_scheme: jit_order.signing_scheme,
+            settlement_contract: Address::default(),
+            sell_token_balance: jit_order.sell_token_balance,
+            buy_token_balance: jit_order.buy_token_balance,
+            presignature_pending: false,
+            pre_interactions: Vec::new(),
+            post_interactions: Vec::new(),
+            ethflow_data: None,
+            onchain_user: None,
+            onchain_placement_error: None,
+            executed_surplus_fee: jit_order.executed_surplus_fee,
+            full_app_data: None,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    pub async fn read_order(
+        ex: &mut PgConnection,
+        uid: &OrderUid,
+    ) -> Result<Option<JitOrder>, sqlx::Error> {
+        const QUERY: &str = r#"
+    SELECT *
+    FROM jit_orders
+    WHERE uid = $1
+        ;"#;
+        sqlx::query_as(QUERY).bind(uid).fetch_optional(ex).await
+    }
+
+    use {
+        super::*,
+        crate::byte_array::ByteArray,
+        sqlx::{Connection, PgConnection},
+    };
+
+    #[tokio::test]
+    #[ignore]
+    async fn postgres_roundtrip() {
+        let mut db = PgConnection::connect("postgresql://").await.unwrap();
+        let mut db = db.begin().await.unwrap();
+        crate::clear_DANGER_(&mut db).await.unwrap();
+
+        // insert a jit order
+        let jit_order = JitOrder {
+            ..Default::default()
+        };
+
+        upsert_orders(&mut db, &[jit_order.clone()]).await.unwrap();
```

Comment:
Please also check the update of the existing order.

Reply:
added